content/c/concepts/union/union.md

```diff
@@ -0,0 +1,92 @@
+---
+Title: 'Union'
+Description: 'A user-defined data type that allows storing different data types in the same memory location.'
+Subjects:
+  - 'Computer Science'
+  - 'Code Foundations'
+Tags:
+  - 'Input'
+  - 'Print'
+  - 'Variables'
+CatalogContent:
+  - 'learn-c'
+  - 'paths/computer-science'
+---
+
+A **union** in C is a special data type that enables a single memory location to hold different types of variables at different times. Unlike structures where each member has its own memory space, all members of a union share the same memory space, and only one member can hold a value at any given time.
+
+## Syntax
+
+```pseudo
+union union_name {
+  data_type member1;
+  data_type member2;
+  ...
+  data_type memberN;
+};
+```
+
+- `union_name`: The name given to the union.
+- `data_type`: It can be any valid C data type (`int`, `float`, `char`, etc.).
+- `member`: The names given to the union members.
+
+## Memory Allocation
+
+In C unions, memory is allocated based on the largest member. For example, if a union contains an integer (4 bytes) and a character (1 byte), the union will occupy 4 bytes. This is because all members share the same memory space – when one member is modified, it affects the memory of other members.
+
+## Accessing Members
+
+Union members can be accessed in two ways: using the dot (.) operator for regular union variables, and the arrow (->) operator for union pointers. However, it is important to remember that only the member that was last assigned a value should be accessed. Since all members share the same memory, accessing an unassigned member will yield unpredictable results.
+
+```c
+union Data {
+    int number;
+    char letter;
+} data;
+
+data.number = 65;
+printf("%d\n", data.number);  // Prints 65
+printf("%c\n", data.letter);  // Prints 'A' (ASCII 65)
+```
+
+The above example sets `data.number = 65` stores the integer value 65 in the shared memory. While `data.letter` is not explicitly set, it shares the same memory as `data.number`, so interpreting the memory as a char results in 'A', the ASCII representation of 65.
```

Comment:
This can be simplified for better readability - 

```suggestion
Here, `data.number = 65` stores the integer 65 in the shared memory. Since `data.letter` uses the same memory, interpreting the value as a character (`char`) results in `'A'`, which is the ASCII representation of 65.
```